rasa/shared/core/domain.py

```diff
@@ -1045,12 +1047,39 @@ def as_dict(self) -> Dict[Text, Any]:
             KEY_INTENTS: self._transform_intents_for_file(),
             KEY_ENTITIES: self._transform_entities_for_file(),
             KEY_SLOTS: self._slot_definitions(),
-            KEY_RESPONSES: self.templates,
+            KEY_RESPONSES: self.get_responses_with_multilines(self.templates),
```

Comment:
I think we should call this in `as_yaml`. What do you think? Feels a bit weird in the regular `as_dict` function which is not necessarily used for dumping.